src/py/flwr_example/quickstart_pytorch/mnist.py

```diff
@@ -0,0 +1,434 @@
+The original code can be inspected in the official PyTorch github:
+
+https://github.com/pytorch/examples/blob/master/mnist/main.py
+"""
+
+
+# mypy: ignore-errors
+# pylint: disable=W0223
+
+import timeit
+from collections import OrderedDict
+from typing import Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor, optim
+from torch.optim.lr_scheduler import StepLR
+from torch.utils.data import DataLoader, Dataset, SubsetRandomSampler
+from torchvision import datasets, transforms
+
+import flwr as fl
+
+
+def dataset_partitioner(
+    dataset: torch.utils.data.Dataset,
+    batch_size: int,
+    client_id: int,
+    number_of_clients: int,
+) -> torch.utils.data.DataLoader:
+    """Helper function to partition datasets
+
+    Parameters
+    ----------
+    dataset: torch.utils.data.Dataset
+        Dataset to be partitioned into *number_of_clients* subsets.
+        
+    batch_size: int 
+        Size of mini-batches used by the returned DataLoader.
+        
+    client_id: int 
+        Unique integer used for selecting a specif partition. 
+        
+    number_of_clients: int 
+        Total number of clients launched during traning. This value dictates the number of partitions to be created.
+        
+
+    Returns
+    -------
+    data_loader: torch.utils.data.Dataset
+        DataLoader for specific client_id considering number_of_clients partitions.  
+
+    """
+
+    # Set the seed so we are sure to generate the same global batches
+    # indices across all clients
+    np.random.seed(123)
+
+    # Get the data corresponding to this client
+    dataset_size = len(dataset)
+    nb_samples_per_clients = dataset_size // number_of_clients
+    dataset_indices = list(range(dataset_size))
+    np.random.shuffle(dataset_indices)
+
+    # Get starting and ending indices w.r.t CLIENT_ID
+    start_ind = client_id * nb_samples_per_clients
+    end_ind = start_ind + nb_samples_per_clients
+    data_sampler = SubsetRandomSampler(dataset_indices[start_ind:end_ind])
+    data_loader = torch.utils.data.DataLoader(
+        dataset, batch_size=batch_size, shuffle=False, sampler=data_sampler
+    )
+    return data_loader
+
+
+def load_data(
+    data_root: str,
+    train_batch_size: int,
+    test_batch_size: int,
+    cid: int,
+    nb_clients: int,
+) -> Tuple[DataLoader, DataLoader]:
+    """Helper function that loads both training and test datasets for MNIST.
+
+    Parameters
+    ----------
+    data_root: str 
+        Directory where MNIST dataset will be stored.
+        
+    train_batch_size: int 
+        Mini-batch size for training set.
+        
+    test_batch_size: int 
+        Mini-batch size for test set.
+        
+    cid: int 
+        Client ID used to select a specific partition.
+        
+    nb_clients: int 
+        Total number of clients launched during traning. This value dictates the number of unique to be created.
```

Comment:
Small typo: `training`

Reply:
Solved in 600102cbbb9579ad7a8add0a14d5e24d9b0f5f7c .